Documentation/MigrationGuide.md

```diff
@@ -0,0 +1,60 @@
+Instead, you can now pass arbitrary parameters to your templates using the `--param X=Y` flag.
+
+All the templates bundled in SwiftGen accept at least the `enumName` parameter to provide at least the same customization as before.
+
+So if you used `swiftgen <command> --enumName Foo` when invoking SwiftGen before, you should now invoke it via `swiftgen <command> --param enumName=Foo`
+
+### You're required to specify a template name or path (no more default template)
+
+Swift evolves so fast that deciding which template should be declared as the default one would require us to change the default template every time a new major version of Swift is released. Besides, our default might not be everyone's default. That's why we decided to not fallback to a default template anymore.
+
+Instead, you're now **required** to provide a template when invoking SwiftGen, either via `-t <templatename>` (you can see the list of available template names using the `swiftgen templates list` command) or via `-p <templatepath>`.
+
+ℹ️ The templates that were used as default template in SwiftGen 4.0 are still bundled with SwiftGen, but they have been renamed (mainly to `"swift2"` since the "default" template in SwiftGen 4 was the one we wrote back in Swift 2 times!) and you now have to explicitly tell which to use. See [below](#templates-bundled-with-swiftgen--some-renaming) for more info.
+
+✅ **Most of you probably already use the `-t swift3` option** if you were writing Swift3 code, so **you won't need to change anything**. But in the unlikely event that you were still writing Swift2 and didn't specify a template, you'll then have to use `-t swift2` to force using this (previously default) template.
+
+## Bundled Templates have been renamed & cleaned
+
+Some templates bundled with SwiftGen have been **renamed** to have a better naming consistency.
+
+Be sure to consult [the templates repo's own Migration Guide](https://github.com/SwiftGen/templates/blob/master/Documentation/MigrationGuide.md) to see the list of changes (renamings or removal) in templates bundled in SwiftGen.
+
+📖 Also, **each template bundled in SwiftGen has a dedicated documentation now**, listing what the template is for, what does the generated code looks like, when you might want to choose that template over another one, and the parameters (`--param X=Y`) supported by this template for customization via the command line.  
+This should help you choose the right template to use for your use case (or help you decide if you need to create your own if not of the provided ones fit your needs) and see what's customizable for each.
+
+📖 See [the documentation folder in the templates repo](https://github.com/SwiftGen/templates/tree/master/Documentation) for those templates documentation. This folder is organized the same way the templates are: one subfolder for each SwiftGen command (`colors`, `strings`, …), then one markdown file for each eponym template.
```

Comment:
I actually don't know if it's used a lot. I know it in French, but idk if it's used a lot in English, and i'm even wondering if that shouldn't be "eponymic" rather than "eponym"…?